### PR TITLE
Add guidance around additional reference requests being cancelled 

### PR DESCRIPTION
**Pull request by davidgisbey**
## Context

At the moment, when a candidate can request as many references as they want. However, when they receive 2 sets of feedback. We cancel all the other references with outstanding feedback(in both the awaiting reference and not requested yet states). 

We don't actually tell the candidate that this is going to happen.

This PR adds in this content (can be found in the Trello ticket)

## Changes proposed in this pull request

- Before

image

- If **no** references are in the not_requested_yet, feedback_requested or feedback_provided state show

image

-  If **one** references are in the not_requested_yet, feedback_requested or feedback_provided state show

image

- If **more than one** 

image

- Does not render if 2 have been received 

## Guidance to review

Is the logic correct around what we consider a viable reference?

I understand that a cancelled reference can be re-requested so in that sense it could technically potentially be viable.

## Link to Trello card

https://trello.com/c/FOoQq16W/2532-%F0%9F%92%942%EF%B8%8F%E2%83%A3-dev-guidance-around-additional-reference-requests-being-cancelled-when-2-have-been-received

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

**Comment by davidgisbey** [comment]
Paul. I thought about localising the strings.

**Comment by paulrobertlloyd** [comment]
> Paul. I can localise the strings, but wanted to get your thoughts as the differences between them are quite subtle and it might obfuscate them. Thought I'd open it up to PR and see what you think.

We currently localise page titles and form text, but there’s a whole other category which we don’t, such as this. Not sure what the solution is to that (or even if one is needed), but don’t think we need to localise the strings here.

File: app/components/candidate_interface/add_reference_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<% unless application_form.enough_references_have_been_provided? %>
+  <div class="govuk-inset-text govuk-!-margin-top-0">
+    <% if enough_references_have_been_added? %>
+
+      <h2 class='govuk-heading-m'>Add another referee</h2>
```
Comment (paulrobertlloyd) [inline app/components/candidate_interface/add_reference_component.html.erb:5]:
```suggestion
      <h2 class="govuk-heading-m">Add another referee</h2>
```

File: app/components/candidate_interface/add_reference_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+      <p class="govuk-body">You can add more referees to increase the chances of getting 2 references quickly.</p>
+
+      <p class="govuk-body">We’ll cancel any remaining requests when you’ve received 2 references.</p>
+
+      <%= link_to 'Add another referee', candidate_interface_references_start_path, class: 'govuk-button' %>
```
Comment (paulrobertlloyd) [inline app/components/candidate_interface/add_reference_component.html.erb:11]: Think it’s possible to do this now:

```suggestion
      <%= govuk_button_link_to 'Add another referee', candidate_interface_references_start_path %>
```

File: app/components/candidate_interface/add_reference_component.html.erb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    <% elsif no_viable_references? %>
+
+      <h2 class='govuk-heading-m'>Add a referee</h2>
```
Comment (paulrobertlloyd) [inline app/components/candidate_interface/add_reference_component.html.erb:15]:
```suggestion
      <h2 class="govuk-heading-m">Add a referee</h2>
```

File: app/components/candidate_interface/add_reference_component.html.erb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+      <p class="govuk-body">You need 2 references before you can submit your application.</p>
+
+      <%= link_to 'Add a referee', candidate_interface_references_start_path, class: 'govuk-button' %>
```
Comment (paulrobertlloyd) [inline app/components/candidate_interface/add_reference_component.html.erb:19]:
```suggestion
      <%= govuk_button_link_to 'Add a referee', candidate_interface_references_start_path %>
```

File: app/components/candidate_interface/add_reference_component.html.erb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+    <% else %>
+
+      <h2 class='govuk-heading-m'>Add a second referee</h2>
```
Comment (paulrobertlloyd) [inline app/components/candidate_interface/add_reference_component.html.erb:23]:
```suggestion
      <h2 class="govuk-heading-m">Add a second referee</h2>
```

File: app/components/candidate_interface/add_reference_component.html.erb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+      <p class="govuk-body">You need 2 references before you can submit your application.</p>
+
+      <%= link_to 'Add a second referee', candidate_interface_references_start_path, class: 'govuk-button' %>
```
Comment (paulrobertlloyd) [inline app/components/candidate_interface/add_reference_component.html.erb:27]:
```suggestion
      <%= govuk_button_link_to 'Add a second referee', candidate_interface_references_start_path %>
```

**Review by paulrobertlloyd — CHANGES_REQUESTED** [review]
I need to test the logic; not sure if viable === given references only. 🤔

**Comment by davidgisbey** [comment]
Made the changes. Sorry I was on autopilot while in a v important meeting. So i think the viable references in this case are:

image

Cancelled/declined/bounced emails shouldn't be counted into their two 'potentially successful' references, but not_requested_yet, received and requested references should be right?

**Comment by gregknight1** [comment]
Content LGTM.

**Comment by paulrobertlloyd** [comment]
This seems to work, just a bit hard to wrap my head around the logic as written. Wonder if it’d be a bit easier to parse if used same terminology across the different scenarios, so `minimum_viable_references` and `no_viable_references`?

**Comment by davidgisbey** [comment]
would it be easier to have 

no_viable_references?
 do x 
one_viable_reference?
  do y 
else 
 do z

**Comment by paulrobertlloyd** [comment]
> would it be easier to have
> 
> no_viable_references?
> do x
> one_viable_reference?
> do y
> else
> do z

Yeah, I think so. The ‘add a second referee’ state is a bit of an outlier.

**Review by paulrobertlloyd — APPROVED** [review]

**Comment by paulrobertlloyd** [comment]
Much clearer – thanks @davidgisbey!

**Pull request merged**